src/main/java/com/powsybl/openloadflow/network/impl/LfHvdcImpl.java

```diff
@@ -125,4 +133,35 @@ public void updateState() {
         ((LfVscConverterStationImpl) converterStation1).getStation().getTerminal().setP(p1.eval() * PerUnit.SB);
         ((LfVscConverterStationImpl) converterStation2).getStation().getTerminal().setP(p2.eval() * PerUnit.SB);
     }
+
+    @Override
+    public boolean isInjectingActiveFlow() {
+        return !isDisabled() && isACEmulationMode;
+    }
+
+    @Override
+    public boolean canTransferActivePower() {
+        // Criteria: if one of the bus is only connected to the HVDC station and nothing else,
+        // no power is transferred. Otherwise the HVDC works as configured
+        return !isolated(bus1) && !isolated(bus2);
+    }
+
+    private boolean isolated(LfBus bus) {
+        if (bus.getGenerators().stream()
+                .filter(g -> !g.isDisabled())
+                .filter(g -> !(g == converterStation1))
+                .anyMatch(g -> !(g == converterStation2))) {
+            return false;
+        }
+        if (bus.getBranches().stream()
+                .anyMatch(b -> !b.isDisabled())
+        ) {
+            return false;
+        }
+        if (!bus.getLoads().isEmpty()) {
```

Comment:
First the condition needs to be checked in the LfNetwork because the topology can change when applying contingencies or operator strategies.
Specifically on loads, it is a translation of the criteria applies at IIM level - we consider the network degenrated if the bus contains only one element, the HDVC station.
In particular we do not want to consider a degenerated network a network that would have one load onthe other side, for example representing a country.
This is the criteria implemented in this PR as discussed internally. But there are other ways to define a degenerated network. This one is simple and should cover basic contingency analysis. But can fail in more complex cases.